src/pages/ExpiringResults/index.js

```diff
@@ -0,0 +1,343 @@
+  selectedControllers: global.selectedControllers,
+  user: user.user,
+  seenResults: user.seenResults,
+  favoriteResults: user.favoriteResults,
+  results: dashboard.results,
+}))
+class ExpiringResults extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      totalResultData: [],
+      toBeDeletedData: [],
+      isModalOpen: false,
+      actionMessage: '',
+    };
+  }
+
+  componentDidMount() {
+    const { dispatch, selectedControllers, selectedDateRange } = this.props;
+    dispatch({
+      type: 'dashboard/fetchResults',
+      payload: {
+        selectedDateRange,
+        controller: selectedControllers,
+      },
+    }).then(() => {
+      const { results } = this.props;
+      this.setState({
+        totalResultData: results[selectedControllers[0] || 'mock-controller'],
+      });
+    });
+  }
+
+  favoriteResult = row => {
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/favoriteResult',
+      payload: row,
+    });
+  };
+
+  unfavoriteResult = row => {
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/removeResultFromFavorites',
+      payload: row,
+    });
+  };
+
+  deleteResult = (e, rows) => {
+    // Stop propagation from going to the next page
+    if (e !== null) {
+      e.stopPropagation();
+    }
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const updatedResult = totalResultData.filter(item => !rows.includes(item.result));
+    dispatch({ type: 'dashboard/updateResults', payload: updatedResult })
+      .then(() => {
+        this.setState({ totalResultData: updatedResult });
+      })
+      .then(() => {
+        this.handleModalToggle(e, []);
+      })
+      .then(() => {
+        this.setState({ actionMessage: 'data succesfully deleted' });
+        setTimeout(() => {
+          this.setState({ actionMessage: '' });
+        }, 3000);
+      });
+  };
+
+  removeResultFromSeen = row => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/removeResultFromSeen',
+      payload: row,
+    });
+  };
+
+  showDrowpdown = (e, id) => {
+    // Stop propagation from going to the next page
+    e.stopPropagation();
+
+    const dropdownElem = document.getElementById(id);
+    if (dropdownElem.style.display === 'none') {
+      dropdownElem.style.display = 'block';
+    } else {
+      dropdownElem.style.display = 'none';
+    }
+  };
+
+  retrieveResults = row => {
+    const { dispatch } = this.props;
+    this.markResultSeen(row);
+    dispatch({
+      type: 'global/updateSelectedResults',
+      payload: [row],
+    }).then(() => {
+      dispatch(
+        routerRedux.push({
+          pathname: '/result',
+        })
+      );
+    });
+  };
+
+  markResultSeen = row => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/markResultSeen',
```

Comment:
Again, it doesn't seem that "seen" should be considered "user" state, but the packaging is of secondary importance here and we can work that out later when we integrate with the server APIs.